TDDWithURMAndKotlin/part3_jump.md

```diff
@@ -0,0 +1,176 @@
+
+2: J(2, 3, 10) -> X: [,3,4] // 4 == 4 so let's move to instruction 10… wait… there's no instruction 10, so we finish the execution
+```
+
+As you can see the `J` a.k.a `jump` function is a simple `go-to` in our machine. It moves the execution from one point to another. We need to create a way to represent this on our interpreter.
+
+## But first… tests
+
+To begin with our `jump` function we need a test for it. We know the function will take a registry to fetch the values to compare and an integer to move the execution to that line. for now, we will pass the set of instructions to the function to allow the `jump` function to move the _current_ index instruction.
+We can take advantage of our other functions already created to setup some registers and then check the function works as expected:
+
+```kotlin
+@Test
+fun `jump function should move to instruction X when registries are equal`() {
+    zero(registry, 5)
+    zero(registry, 10)
+
+    jump(registry, 5, 10, instructionSet, 1)
+
+    assertEquals(1, instructionSet.current)
+}
+```
+
+Wow, that's actually an ugly function. It takes 5 parameters, probably Uncle Bob is looking at us and feeling disappointed… well, we will refactor it later. For now let's see if it works… well, actually we know it doesn't work, as [we see previously](https://dev.to/sierisimo/tdd-with-urm-and-kotlin-1dj7), if we run `gradle check` to run the tests, it will generate a _compile-time error_ because we haven't created the `jump` function yet.
+
+Let's fix that:
+
+```kotlin
+fun jump(registry: Registry, positionX: Int, positionY: Int, instructionSet: ?, instruction: Int) {
+
+}
+```
+
+Before moving on, we need to define how our `instrcutionSet` will behave or what type of data will be. Things to keep in mind:
+
+- It should hold as much instructions as need it
+- Needs to know which is the current instruction as which one is the next one based on the index
+- Needs to easily change the pointer of execution to any other instruction
+
+We could need more functionality from the `instructionSet` but having this 3 requirements tell us we need a class and will let us at least write the minimum necessary to fullfill our failing test. First we need to change the test adding a type for `instructionSet`:
+
+```kotlin
+internal class OperationsTest {
+    //…
+
+    val instructionSet = InstructionSet()
+}
+```
+
+And we can use it in the test, but wait, the test requires the `instructionSet` to have a `current` property. Let's create the class and add the property to remove the _compile-time error_:
+
+```kotlin
+class InstructionSet {
+    var current: Int = 0
+}
+```
+
+We use `var` because we will be setting the value to a different number on every execution of the instruction set. This class just holds data and actually doesn't need to do anything else for our current porpouse. Our test should pass now right? No compile time errors and it's quite straight forward.
```

Comment:
```suggestion
We use `var` because we will be setting the value to a different number on every execution of the instruction set. This class just holds data and actually doesn't need to do anything else for our current purpose. Our test should pass now, right? No compile time errors and it's quite straightforward.
```
Typos~